pkg/trace/obfuscate/sql.go

```diff
@@ -195,20 +200,41 @@ type tableFinderFilter struct {
 // Filter implements tokenFilter.
 func (f *tableFinderFilter) Filter(token, lastToken TokenKind, buffer []byte) (TokenKind, []byte, error) {
 	switch lastToken {
-	case From:
+	case From, Join:
 		// SELECT ... FROM [tableName]
 		// DELETE FROM [tableName]
+		// ... JOIN [tableName]
 		if r, _ := utf8.DecodeRune(buffer); !unicode.IsLetter(r) {
 			// first character in buffer is not a letter; we might have a nested
 			// query like SELECT * FROM (SELECT ...)
 			break
 		}
 		fallthrough
-	case Update, Into, Join:
+	case Update, Into:
 		// UPDATE [tableName]
 		// INSERT INTO [tableName]
-		// ... JOIN [tableName]
-		f.storeName(string(buffer))
+		if f.storeNames {
+			f.storeName(string(buffer))
+		}
+
+		if f.normalizeTables && token == ID {
+			// Replace all digits
+			buf := make([]byte, 0, len(buffer))
```

Comment:
Would `strings.Builder` be faster here? Can we try it out? We could put this logic in a function and benchmark it.

Reply:
It appears to be worse:

```
Current:
BenchmarkTableNormalizer/#00-8          18532640          65.4 ns/op        32 B/op        1 allocs/op

StringBuilder:
BenchmarkTableNormalizer/#00-8         	 7361636	       163 ns/op	      56 B/op	       3 allocs/op
```

```go
func (f *replaceFilter) replaceTableDigits(buffer []byte) string {
	// Replace all digits
	// buf := make([]byte, 0, len(buffer))
	var sb strings.Builder
	scanningDigit := false
	for _, c := range string(buffer) {
		if isDigit(c) {
			if scanningDigit {
				continue
			}
			scanningDigit = true
			// buf = append(buf, byte('?'))
			sb.WriteRune('?')
			continue
		}
		scanningDigit = false
		// buf = append(buf, byte(c))
		sb.WriteRune(c)
	}
	return sb.String()
}
```